System/src/Check.cs

```diff
@@ -241,70 +246,115 @@ public static string ThrowIfNullOrWhitespace<T>(this string? value, string messa
+	// Throw if value is not equal to expected
+	
 	public static bool ThrowIfNotEqual(this int value, int expected)
 		=> value.ThrowIfNotEqual<ArgumentNotEqualException>(expected, nameof(value));
 
-	[ContractAnnotation(AnnotationResources.ValueNullThenHalt)]
+	public static bool ThrowIfNotEqual<T>(this int value, int expected)
+		where T : ArgumentException
+		=> value.ThrowIfNotEqual<T>(expected, nameof(value));
+
 	public static bool ThrowIfNotEqual<T>(this int value, int expected, [InvokerParameterName] string parameterName)
 		where T : ArgumentException
 		=> value != expected
 			   ? throw CreateExceptionInstance<T>(parameterName)
 			   : true;
+	
+	// Throw if value is less than expected
 
-	[ContractAnnotation(AnnotationResources.ValueNullThenHalt)]
 	public static bool ThrowIfLessThan(this int value, int expected)
-		=> ThrowIfLessThan(value, expected, nameof(value));
-
-	[ContractAnnotation(AnnotationResources.ValueNullThenHalt)]
+		=> value.ThrowIfLessThan<ArgumentLessThanException>(expected, nameof(value));
+	
 	public static bool ThrowIfLessThan(this int value, int expected, [InvokerParameterName] string parameterName)
+		=> value.ThrowIfLessThan<ArgumentLessThanException>(expected, parameterName);
+	
+	public static bool ThrowIfLessThan<T>(this int value, int expected)
+		where T : ArgumentException
+		=> value.ThrowIfLessThan<T>(expected, nameof(value));
+	
+	public static bool ThrowIfLessThan<T>(this int value, int expected, [InvokerParameterName] string parameterName)
+		where T : ArgumentException
 		=> value < expected
-			   ? throw new ArgumentLessThanException(parameterName)
+			   ? throw CreateExceptionInstance<T>(parameterName)
 			   : true;
-
-	[ContractAnnotation(AnnotationResources.ValueNullThenHalt)]
+	
+	// Throw if value is more than expected
+	
 	public static bool ThrowIfMoreThan(this int value, int expected)
-		=> ThrowIfMoreThan(value, expected, nameof(value));
+		=> value.ThrowIfMoreThan<ArgumentMoreThanException>(expected, nameof(value));
+	
+	public static bool ThrowIfMoreThan(this int value, int expected, [InvokerParameterName] string parameterName)
+		=> value.ThrowIfMoreThan<ArgumentMoreThanException>(expected, parameterName);
 
-	[ContractAnnotation(AnnotationResources.ValueNullThenHalt)]
-	internal static bool ThrowIfMoreThan(this int value, int expected, [InvokerParameterName] [NotNull] string parameterName)
+
+	public static bool ThrowIfMoreThan<T>(this int value, int expected)
+		where T : ArgumentException
+		=> value.ThrowIfMoreThan<T>(expected, nameof(value));
+
+
+	public static bool ThrowIfMoreThan<T>(this int value, int expected, [InvokerParameterName] string parameterName)
+		where T : ArgumentException
 		=> value > expected
-			   ? throw new ArgumentMoreThanException(parameterName)
+			   ? throw CreateExceptionInstance<T>(parameterName)
 			   : true;
+
+	// Throw if value is zero
 	
+	public static bool ThrowIfZero(this int value)
```

Comment:
add the throw if zero

Reply:
Maybe having a checking case for `ThrowIfPossitive()` and `ThrowIfNegative()` would be helpful. I will open a new issue to track this idea.